File: UI/app/src/main/java/com/rikkei/training/ui/dialog/NumberPickerDialog.java
package com.rikkei.training.ui.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.NumberPicker;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;

import com.rikkei.training.ui.R;
import com.rikkei.training.ui.base.BaseDialogFragment;
import com.rikkei.training.ui.databinding.DialogMessageLimitBinding;

public class NumberPickerDialog extends BaseDialogFragment {
    private DialogMessageLimitBinding binding;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.dialog_message_limit, null, false);
        binding.numberPicker.setMinValue(1);
        binding.numberPicker.setMaxValue(8);

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity())
                .setTitle(R.string.number_picker_dialog)
                .setView(binding.getRoot())
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                })
                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                });
        return builder.create();
    }

    public NumberPickerDialog(Context context) {
        super(context);
    }
}

File: UIKotlin/app/src/main/java/com/rikkei/training/uikotlin/dialog/NumberPickerDialog.kt
package com.rikkei.training.uikotlin.dialog


import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.NumberPicker
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import com.rikkei.training.uikotlin.R
import com.rikkei.training.uikotlin.base.BaseDialogFragment
import com.rikkei.training.uikotlin.databinding.DialogMessageLimitBinding
import kotlinx.android.synthetic.*
import kotlinx.android.synthetic.main.dialog_message_limit.*


class NumberPickerDialog(context: Context) : BaseDialogFragment(context) {
    private var binding: DialogMessageLimitBinding? = null
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = DataBindingUtil.inflate(inflater, R.layout.dialog_message_limit, null, false);
        binding!!.numberPicker.minValue = 1
        binding!!.numberPicker.maxValue = 8

        val builder = AlertDialog.Builder(activity)
            .setTitle(R.string.number_picker_dialog)
            .setView(binding!!.root)
            .setPositiveButton(
                R.string.ok
            ) { _, _ -> }
            .setNegativeButton(
                R.string.cancel
            ) { _, _ -> }
        return builder.create()
    }
}
